### DISCUSSION TITLE
使用antd组件自带的方法怎么写ts

### DISCUSSION THREAD

**Discussion by dddssw** [ant-design/ant-design#48250] [Q&A]
` <Form form={form} name="basic" layout="vertical" onFinish={onFinish} autoComplete="off" onFieldsChange={watchDataChange} style={{ background: '#fff', padding: '20px' }}>`

` function watchDataChange(changedFields, allFields) { }`

现在changedFields，allFields有ts报错Parameter 'allFields' implicitly has an 'any' type

如何让他不要报错

**Comment by zombieJ** [comment]
```tsx
const watchDataChange: FormProps<RecordType>[onFieldsChange'] = xxx
```

**Reply by dddssw** [reply]
谢谢，另外这个报错类型是从哪来的
![image](https://github.com/ant-design/ant-design/assets/58782768/c77a27fe-8336-4475-b8ec-0f6bdea1fca3)

**Reply by kiner-tang** [reply]
RecordType 是一个泛型参数来的，你需要根据你的表单对象的类型传入对应的泛型，例如你的表单对象是：`type UserInfo = {name: string, age: number}`,那么你这边就应该是：
```tsx
const watchDataChange: FormProps<UserInfo>[onFieldsChange'] = xxx
```
